src/Catalyst.Core.Lib/Cryptography/KeyChain.cs

```diff
@@ -0,0 +1,472 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Core.Lib.Config;
+using Catalyst.Core.Lib.FileSystem;
+using Common.Logging;
+using MultiFormats;
+using Org.BouncyCastle.Asn1.Sec;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Digests;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Math;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.X509;
+
+namespace Catalyst.Core.Lib.Cryptography
+{
+    /// <summary>
+    ///   A secure key chain.
+    /// </summary>
+    public partial class KeyChain : IKeyApi
+    {
+        private static ILog log = LogManager.GetLogger(typeof(KeyChain));
+
+        private char[] dek;
+        private FileStore<string, EncryptedKey> store;
+
+        /// <summary>
+        ///   Create a new instance of the <see cref="KeyChain"/> class.
+        /// </summary>
+        public KeyChain() { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this redundant constructor.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)